supreme/src/commonMain/kotlin/at/asitplus/signum/supreme/validate/CertificateChainValidator.kt

```diff
@@ -0,0 +1,172 @@
+package at.asitplus.signum.supreme.validate
+
+import at.asitplus.signum.CertificateChainValidatorException
+import at.asitplus.signum.CryptoOperationFailed
+import at.asitplus.signum.KeyUsageException
+import at.asitplus.signum.indispensable.asn1.KnownOIDs
+import at.asitplus.signum.indispensable.asn1.ObjectIdentifier
+import at.asitplus.signum.indispensable.pki.CertificateChain
+import at.asitplus.signum.indispensable.pki.X509Certificate
+import at.asitplus.signum.indispensable.pki.X509KeyUsage
+import at.asitplus.signum.indispensable.pki.root
+import at.asitplus.signum.supreme.sign.verifierFor
+import at.asitplus.signum.supreme.sign.verify
+import kotlinx.datetime.Clock
+import kotlinx.datetime.Instant
+
+val supportedCriticalExtensionOids = setOf(
+    KnownOIDs.keyUsage,
+    KnownOIDs.certificatePolicies_2_5_29_32,
+    KnownOIDs.policyMappings,
+    KnownOIDs.inhibitAnyPolicy,
+    KnownOIDs.cRLDistributionPoints_2_5_29_31,
+    KnownOIDs.issuingDistributionPoint_2_5_29_28,
+    KnownOIDs.deltaCRLIndicator,
+    KnownOIDs.policyConstraints_2_5_29_36,
+    KnownOIDs.basicConstraints_2_5_29_19,
+    KnownOIDs.subjectAltName_2_5_29_17,
+    KnownOIDs.nameConstraints_2_5_29_30
+)
+
+sealed interface Validator {
+    fun check(currCert: X509Certificate)
```

Comment:
Could this be made suspend, so it's more flexible and can e.g. do requests if necessary?